Eliminate use of getPropertyCSSValue from tests, part 2

darinadler opened the pull request:
#### c710b3e74570dd69487ac9b9a72ebc1ffb0f80f3
<pre>
Eliminate use of getPropertyCSSValue from tests, part 2
<a href="https://bugs.webkit.org/show_bug.cgi?id=293590">https://bugs.webkit.org/show_bug.cgi?id=293590</a>
<a href="https://rdar.apple.com/152048914">rdar://152048914</a>

Reviewed by Anne van Kesteren and Sam Weinig.

Some of this is removing possibly-useful test coverage; we should be cautious about that.
See comments in <a href="https://github.com/WebKit/WebKit/pull/45904">https://github.com/WebKit/WebKit/pull/45904</a>

* LayoutTests/TestExpectations: Expect transitions/svg-text-shadow-transition.html to pass.
There is some risk that it will now be flaky, and we&apos;ll have to add another
flakiness expectation like the ones from webkit.org/b/137883.

* LayoutTests/animations/resources/animation-test-helpers.js:
(getPropertyValue): Rewrote this to have fewer special cases and not use
getPropertyCSSValue at all.

* LayoutTests/fast/css/CSSStyleDeclaration-parameters-expected.txt: Updated.
* LayoutTests/fast/css/CSSStyleDeclaration-parameters.html: Test getPropertyValue
instead of getPropertyCSSValue.

* LayoutTests/fast/css/font-systemFontID-parsing-expected.txt: Updated.
* LayoutTests/fast/css/font-systemFontID-parsing.html: Test getPropertyValue instead
of getPropertyCSSValue. Use js-test.js.

* LayoutTests/fast/css/getComputedStyle/computed-style-font-expected.txt: Updated.
* LayoutTests/fast/css/getComputedStyle/computed-style-font.html: Removed the tests of
getPropertyCSSValue. Simplified the tests that were left behind.

* LayoutTests/fast/css/getComputedStyle/getComputedStyle-background-position-expected.txt:
Updated.
* LayoutTests/fast/css/getComputedStyle/getComputedStyle-background-position.html:
Removed the test of getPropertyCSSValue.

* LayoutTests/fast/css/getComputedStyle/getComputedStyle-background-shorthand-expected.txt:
Updated.
* LayoutTests/fast/css/getComputedStyle/getComputedStyle-background-shorthand.html:
Removed the tests of getPropertyCSSValue. Simplified the tests that were left behind.

* LayoutTests/fast/css/getComputedStyle/getComputedStyle-background-size-expected.txt:
Updated.
* LayoutTests/fast/css/getComputedStyle/getComputedStyle-background-size.html:
Removed the tests of getPropertyCSSValue. Simplified the tests that were left behind.

* LayoutTests/fast/css/getComputedStyle/getComputedStyle-border-color-shorthand-expected.txt:
Updated.
* LayoutTests/fast/css/getComputedStyle/getComputedStyle-border-color-shorthand.html:
Removed the tests of getPropertyCSSValue. Simplified the tests that were left behind.

* LayoutTests/fast/css3-text/css3-text-decoration/text-decoration-skip/text-decoration-skip-roundtrip-expected.txt:
Updated.
* LayoutTests/fast/css3-text/css3-text-decoration/text-decoration-skip/text-decoration-skip-roundtrip.html:
Test with getPropertyValue instead of getPropertyCSSValue.

* LayoutTests/fast/css3-text/font-synthesis-parse-expected.txt: Updated.
* LayoutTests/fast/css3-text/font-synthesis-parse.html: Test with getPropertyValue
instead of getPropertyCSSValue.

* LayoutTests/platform/glib/transitions/svg-text-shadow-transition-expected.txt: Removed.
* LayoutTests/platform/gtk/transitions/svg-text-shadow-transition-expected.png: Removed.
* LayoutTests/platform/ios/transitions/svg-text-shadow-transition-expected.txt: Removed.
* LayoutTests/platform/mac/transitions/svg-text-shadow-transition-expected.png: Removed.
* LayoutTests/platform/mac/transitions/svg-text-shadow-transition-expected.txt: Removed.
* LayoutTests/platform/win/transitions/svg-text-shadow-transition-expected.txt: Removed.

* LayoutTests/transitions/frames-timing-function-expected.txt: Updated expected
failure messages since the test helper now helpfully cites the single value instead
of the whole matrix.

* LayoutTests/transitions/lengthsize-transition-to-from-auto-expected.txt: Updated.
* LayoutTests/transitions/lengthsize-transition-to-from-auto.html: Added an expected
value of &quot;auto&quot; since the test helper no longer ignores its presence.

* LayoutTests/transitions/resources/transition-test-helpers.js: Rewrote the checking
code to use the expected value types to decide what kind of comparision to do and use
the string computed style value rather than using the result of getPropertyCSSValue.

* LayoutTests/transitions/svg-text-shadow-transition-expected.txt: Added.
This is now a text based text rather than a pixel test.
* LayoutTests/transitions/svg-text-shadow-transition.html: Change expectations to
match the serialization of text-shadow, including the color.

* LayoutTests/transitions/transition-to-from-auto.html: Expect the number 50 rather
than the string &quot;50&quot; now that the test helper uses the expected type to figure out
how to check values. Alternatively could have expected the string &quot;50px&quot;, but this
preserves the old behavior and the tolerance.

* LayoutTests/transitions/transition-to-from-undefined.html: Expect the numbers 100
and 50 rather than the strings &quot;100&quot; and &quot;50&quot; now that the test helper uses the expected
type to figure out how to check values. Alternatively could have expected the strings
&quot;100px&quot; and &quot;50px&quot;, but this preserves the old behavior and the tolerance.

Canonical link: <a href="https://commits.webkit.org/295994@main">https://commits.webkit.org/295994@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/db489546e1877204a85b8fea960426f882e2950f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/106818 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/26569 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/16967 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/112029 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/57416 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/27239 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/35071 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/81110 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/109822 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/21624 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/96392 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/61451 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/21064 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/14500 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/56870 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/90965 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/14528 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/114984 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/33956 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/25048 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/90172 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/34321 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/92623 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/89882 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/34822 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/12638 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/29640 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/17297 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/33881 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/33628 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/36981 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/35226 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

annevk commented on LayoutTests/animations/resources/animation-test-helpers.js:
```diff
@@ -416,43 +416,37 @@ function checkExpectedValue(expected, index)
+    ])
+    if (!nonNumericPropertyNames.has(propertyPrefix))
+        return parseFloat(value)
+    return value
```
FWIW: removing all the semi-colons is generally frowned upon in JavaScript as there are cases where automatic semi-colon insertion can lead to problems.

darinadler replied:
I did not know that! I will add semicolons back

annevk reviewed (approved):
This looks fine, but maybe revisit the choice on semi-colon removal.

darinadler commented on LayoutTests/transitions/svg-text-shadow-transition.html:
```diff
@@ -21,17 +21,16 @@
             // [time, element-id, property, expected-value, tolerance]
-            [0, 'text', 'text-shadow', [-25, -5, 0, 15], 4],
-            [1, 'text', 'text-shadow', [5, 5, 0, 15], 4]
+            [0, 'text', 'text-shadow', [[254, 0, 0], -25, -5, 15], 4],
```
This should be 255, not 254. But also, I need to figure out why the tolerance of 4 isn’t close enough to match in WPE tests:
 ```
FAIL - "text-shadow" property for "text" element at 0s expected: 254,0,0,-25,-5,15 but saw: rgb(255, 0, 0) -24.969999px -4.99px 15px
FAIL - "text-shadow" property for "text" element at 1s expected: 0,128,0,5,5,15 but saw: rgb(3, 126, 0) 4.61px 4.87px 15px
```

darinadler committed:
Eliminate use of getPropertyCSSValue from tests, part 2
https://bugs.webkit.org/show_bug.cgi?id=293590
rdar://152048914

Reviewed by Anne van Kesteren and Sam Weinig.

Some of this is removing possibly-useful test coverage; we should be cautious about that.
See comments in https://github.com/WebKit/WebKit/pull/45904

* LayoutTests/TestExpectations: Expect transitions/svg-text-shadow-transition.html to pass.
There is some risk that it will now be flaky, and we'll have to add another
flakiness expectation like the ones from webkit.org/b/137883.

* LayoutTests/animations/resources/animation-test-helpers.js:
(getPropertyValue): Rewrote this to have fewer special cases and not use
getPropertyCSSValue at all.

* LayoutTests/fast/css/CSSStyleDeclaration-parameters-expected.txt: Updated.
* LayoutTests/fast/css/CSSStyleDeclaration-parameters.html: Test getPropertyValue
instead of getPropertyCSSValue.

* LayoutTests/fast/css/font-systemFontID-parsing-expected.txt: Updated.
* LayoutTests/fast/css/font-systemFontID-parsing.html: Test getPropertyValue instead
of getPropertyCSSValue. Use js-test.js.

* LayoutTests/fast/css/getComputedStyle/computed-style-font-expected.txt: Updated.
* LayoutTests/fast/css/getComputedStyle/computed-style-font.html: Removed the tests of
getPropertyCSSValue. Simplified the tests that were left behind.

* LayoutTests/fast/css/getComputedStyle/getComputedStyle-background-position-expected.txt:
Updated.
* LayoutTests/fast/css/getComputedStyle/getComputedStyle-background-position.html:
Removed the test of getPropertyCSSValue.

* LayoutTests/fast/css/getComputedStyle/getComputedStyle-background-shorthand-expected.txt:
Updated.
* LayoutTests/fast/css/getComputedStyle/getComputedStyle-background-shorthand.html:
Removed the tests of getPropertyCSSValue. Simplified the tests that were left behind.

* LayoutTests/fast/css/getComputedStyle/getComputedStyle-background-size-expected.txt:
Updated.
* LayoutTests/fast/css/getComputedStyle/getComputedStyle-background-size.html:
Removed the tests of getPropertyCSSValue. Simplified the tests that were left behind.

* LayoutTests/fast/css/getComputedStyle/getComputedStyle-border-color-shorthand-expected.txt:
Updated.
* LayoutTests/fast/css/getComputedStyle/getComputedStyle-border-color-shorthand.html:
Removed the tests of getPropertyCSSValue. Simplified the tests that were left behind.

* LayoutTests/fast/css3-text/css3-text-decoration/text-decoration-skip/text-decoration-skip-roundtrip-expected.txt:
Updated.
* LayoutTests/fast/css3-text/css3-text-decoration/text-decoration-skip/text-decoration-skip-roundtrip.html:
Test with getPropertyValue instead of getPropertyCSSValue.

* LayoutTests/fast/css3-text/font-synthesis-parse-expected.txt: Updated.
* LayoutTests/fast/css3-text/font-synthesis-parse.html: Test with getPropertyValue
instead of getPropertyCSSValue.

* LayoutTests/platform/glib/transitions/svg-text-shadow-transition-expected.txt: Removed.
* LayoutTests/platform/gtk/transitions/svg-text-shadow-transition-expected.png: Removed.
* LayoutTests/platform/ios/transitions/svg-text-shadow-transition-expected.txt: Removed.
* LayoutTests/platform/mac/transitions/svg-text-shadow-transition-expected.png: Removed.
* LayoutTests/platform/mac/transitions/svg-text-shadow-transition-expected.txt: Removed.
* LayoutTests/platform/win/transitions/svg-text-shadow-transition-expected.txt: Removed.

* LayoutTests/transitions/frames-timing-function-expected.txt: Updated expected
failure messages since the test helper now helpfully cites the single value instead
of the whole matrix.

* LayoutTests/transitions/lengthsize-transition-to-from-auto-expected.txt: Updated.
* LayoutTests/transitions/lengthsize-transition-to-from-auto.html: Added an expected
value of "auto" since the test helper no longer ignores its presence.

* LayoutTests/transitions/resources/transition-test-helpers.js: Rewrote the checking
code to use the expected value types to decide what kind of comparision to do and use
the string computed style value rather than using the result of getPropertyCSSValue.

* LayoutTests/transitions/svg-text-shadow-transition-expected.txt: Added.
This is now a text based text rather than a pixel test.
* LayoutTests/transitions/svg-text-shadow-transition.html: Change expectations to
match the serialization of text-shadow, including the color.

* LayoutTests/transitions/transition-to-from-auto.html: Expect the number 50 rather
than the string "50" now that the test helper uses the expected type to figure out
how to check values. Alternatively could have expected the string "50px", but this
preserves the old behavior and the tolerance.

* LayoutTests/transitions/transition-to-from-undefined.html: Expect the numbers 100
and 50 rather than the strings "100" and "50" now that the test helper uses the expected
type to figure out how to check values. Alternatively could have expected the strings
"100px" and "50px", but this preserves the old behavior and the tolerance.

Canonical link: https://commits.webkit.org/295994@main
